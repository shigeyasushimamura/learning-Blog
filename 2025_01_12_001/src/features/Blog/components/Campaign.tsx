import styles from "./Blog.module.css";

const Campaign = () => {
  return (
    <>
      <div className={styles["campaign"]}>
        <img
          className={styles["campaign__img"]}
          src="fv_pc.webp"
          alt="キャンペーンメインビジュアル"
        ></img>

        <div className={styles["campaign__img-container"]}>
          <a
            href="/"
            className={styles["campaign__community-open"]}
            aria-label="コミュニティを開く"
          >
            <img
              className={styles["campaign__img-community"]}
              src="community_pc.webp"
              alt="コミュニティ参加案内"
            ></img>
          </a>
        </div>

        <div className={styles["campaign__container"]}>
          <button
            className={styles["campaign__btn"]}
            aria-label="体験授業の申し込み"
          >
            無料で体験授業を受ける
          </button>
        </div>

        <div className={styles["campaign__container"]}>
          <a>
            <img
              src="banner.webp"
              alt="バナー"
              className={`${styles["campaign__img"]} ${styles["img__transparent"]}`}
            ></img>
          </a>
        </div>

        <div className={styles["campaign__container"]}>
          <img
            src="accent.webp"
            alt="アクセント"
            className={styles["campaign__img-accent"]}
          ></img>
        </div>
      </div>
    </>
  );
};

export default Campaign;
