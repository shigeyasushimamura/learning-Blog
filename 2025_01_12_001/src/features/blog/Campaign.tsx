import styles from "./Blog.module.css";

const Campaign = () => {
  return (
    <>
      <div className={styles["campaign"]}>
        <img
          className={styles["campaign__img"]}
          src="fv_pc.webp"
          alt="キャンペーンメインビジュアル"
        ></img>

        <div className={styles["campaign__img-container"]}>
          <a
            href="/"
            className={styles["campaign__community-open"]}
            aria-label="コミュニティを開く"
          >
            <img
              className={styles["campaign__img"]}
              src="community_pc.webp"
              alt="コミュニティ参加案内"
            ></img>
          </a>
        </div>

        <div className={styles["campaign__btn-container"]}>
          <button
            className={styles["campaign__btn"]}
            aria-label="体験授業の申し込み"
          >
            無料で体験授業を受ける
          </button>
        </div>
      </div>
    </>
  );
};

export default Campaign;
