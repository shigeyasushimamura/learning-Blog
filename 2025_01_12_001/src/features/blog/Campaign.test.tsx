import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import Campaign from "./Campaign";
import styles from "./Blog.module.css";
import "@testing-library/jest-dom";

vi.mock("./Blog.module.css", () => ({
  default: {
    campaign: "campaign",
    campaign__img: "campaign__img",
    campaign__img_container: "campaign__img-container",
    campaign__community_open: "campaign__community-open",
    campaign__btn_container: "campaign__btn-container",
    campaign__btn: "campaign__btn",
  },
}));

describe("Campaign", () => {
  it("renders main campaign image with correct attributes", () => {
    const { asFragment } = render(<Campaign />);

    expect(asFragment()).toMatchSnapshot();
  });
});
