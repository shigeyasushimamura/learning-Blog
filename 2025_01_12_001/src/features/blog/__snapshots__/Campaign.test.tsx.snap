// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Campaign > renders main campaign image with correct attributes 1`] = `
<DocumentFragment>
  <div
    class="campaign"
  >
    <img
      alt="キャンペーンメインビジュアル"
      class="campaign__img"
      src="fv_pc.webp"
    />
    <div>
      <a
        aria-label="コミュニティを開く"
        href="/"
      >
        <img
          alt="コミュニティ参加案内"
          class="campaign__img"
          src="community_pc.webp"
        />
      </a>
    </div>
    <div>
      <button
        aria-label="体験授業の申し込み"
        class="campaign__btn"
      >
        無料で体験授業を受ける
      </button>
    </div>
  </div>
</DocumentFragment>
`;
